lib/jsonapi/query_builder/mixins/sort.rb

```diff
@@ -76,7 +89,11 @@ def sort_params
         end
 
         def ensure_permitted_sort_params!(sort_params)
-          unpermitted_parameters = sort_params.map(&:attribute).map(&:to_sym) - self.class.supported_sorts.keys
+          unpermitted_parameters = sort_params.map(&:attribute).filter do |attribute|
+            next false if self.class.supported_sorts.key?(attribute.to_sym)
+
+            self.class.supported_dynamic_sorts.none? { |dynamic_sort| dynamic_sort.matches?(attribute) }
```

Comment:
Optional, I find it easier to understand written in the following manner
```suggestion
            !self.class.supported_sorts.key?(attribute.to_sym) &&
              self.class.supported_dynamic_sorts.none? { |dynamic_sort| dynamic_sort.matches?(attribute) }
```